### PR TITLE
feat(core): routes to toggle blocklist

### PR DESCRIPTION
**Pull request by prajjwalkumar17**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This Api mentioned will be for toggling the `blocklist-guard` for a particular merchant. So the changes will be as mentioned:

> It will try to find a config associated with that merchant (guard_blocklist_for_{merchant_id}) which in tern is a boolean config.
> If present it will enable/disable it accordingly.
> If not present it will create a config and assign it appropriate value.

More in the linked issue.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

This can be found out in the linked issue.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/routes/blocklist.rs [hunk 1]
```diff
@@ -117,3 +117,41 @@ pub async fn list_blocked_payment_methods(
...
+        ("data_kind" = BlocklistDataKind, Query, description = "Kind of the fingerprint list requested"),
+    ),
+    responses(
+        (status = 200, description = "Blocked Fingerprints", body = BlocklistResponse),
+        (status = 400, description = "Invalid Data")
+    ),
+    tag = "Blocklist",
+    operation_id = "List Blocked fingerprints of a particular kind",
+    security(("api_key" = []))
+)]
```
Comment (prajjwalkumar17) [inline crates/router/src/routes/blocklist.rs:134]: need to refactor this accordingly.

File: crates/router/src/core/blocklist/utils.rs [hunk 2]
```diff
@@ -85,6 +86,54 @@ pub async fn delete_entry_from_blocklist(
...
+            state
+                .store
+                .insert_config(new_config)
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error enabling the blocklist guard")?;
+        }
+        Err(e) => {
+            logger::error!(error=?e);
+            return Err(errors::ApiErrorResponse::InternalServerError.into());
```
Comment (Narayanbhat166) [inline crates/router/src/core/blocklist/utils.rs:122]:
```suggestion
            Err(e).change_context(errors::ApiErrorResponse::InternalServerError)
```

File: crates/router/src/routes/app.rs [hunk 3]
```diff
@@ -658,6 +658,9 @@ impl Blocklist {
                     .route(web::post().to(blocklist::add_entry_to_blocklist))
                     .route(web::delete().to(blocklist::remove_entry_from_blocklist)),
             )
+            .service(
+                web::resource("/enable").route(web::post().to(blocklist::toggle_blocklist_guard)),
```
Comment (Narayanbhat166) [inline crates/router/src/routes/app.rs:662]: shouldn't this be toggle?

File: crates/router/src/core/blocklist/utils.rs [hunk 4]
```diff
@@ -85,6 +86,56 @@ pub async fn delete_entry_from_blocklist(
...
+            logger::error!(error=?e);
+            Err(e)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error enabling the blocklist guard")?;
+        }
+    };
+    let guard_status = if query.status { "enabled" } else { "disabled" };
+    Ok(api_blocklist::ToggleBlocklistResponse {
+        blocklist_guard_status: format!("{} {}", "blocklist guard is", guard_status),
+    })
```
Comment (vspecky) [inline crates/router/src/core/blocklist/utils.rs:130]:
```suggestion
    Ok(api_blocklist::ToggleBlocklistResponse {
        blocklist_guard_status: guard_status,
    })
```
No need to send back a message. If the merchant wants to make a decision based on the response, it's easier to match on the standalone status as opposed to a message.

**Review by vspecky — CHANGES_REQUESTED** [review]

**Review by vspecky — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**